src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -181,12 +213,12 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
         return this._adyenv2;
     }
 
-    private _getStateContainer(): string {
-        if (!this._stateContainer) {
-            return '{}';
+    private _getPaymentMethodState(): AdyenCardState {
+        if (!this._paymentMethodState) {
+            throw new InvalidArgumentError('"stateContainer" argument was not provided during initialization.');
```

Comment:
This is not the reason why `_paymentMethodState` is undefined. There's no way for the user of the SDK to fix this problem because `stateContainer` is not an argument that they can provide during initialisation. It is just a private property of the class that you've defined.

I think I've said this before. Defining a getter like this is an anti-pattern because it prevents you from throwing an appropriate error. The error you throw depends on the context of the call. And sometimes, it's fine to not throw an error because it's in the expected state. You should avoid doing this in the future.

In this particular case, it seems you might not be able to proceed with submitting the payment if the `execute` method is called before the shopper has interacted with the payment component and selected a card (before the `onChange` is called). So at the beginning of the `execute` method, you can put a guard statement to stop the code execution unless `_paymentMethodState` is defined.

```ts
const paymentMethodState = this._paymentMethodState;

if (!paymentMethodState) {
    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
}
```